src/OpenApi/src/Services/OpenApiDocumentProvider.cs

```diff
@@ -25,7 +25,9 @@ internal sealed class OpenApiDocumentProvider(IServiceProvider serviceProvider)
     /// <param name="writer">A text writer associated with the document to write to.</param>
     public async Task GenerateAsync(string documentName, TextWriter writer)
     {
-        var optionsSnapshot = serviceProvider.GetRequiredService<IOptionsSnapshot<OpenApiOptions>>();
+        // Resolving IOptionsSnapshot requires a scoped service provider.
+        using var scopedService = serviceProvider.CreateScope();
+        var optionsSnapshot = scopedService.ServiceProvider.GetRequiredService<IOptionsSnapshot<OpenApiOptions>>();
         var namedOption = optionsSnapshot.Get(documentName);
```

Comment:
```suggestion
        var optionsMonitor = serviceProvider.GetRequiredService<IOptionsSnapshot<OpenApiOptions>>();
        var namedOption = optionsMonitor.Get(documentName);
```

It would probably be cleaner to use `IOptionsMonitor<>` instead of creating a scope. We often use it without registering for change notifications since it supports named options. We don't really use `IOptionsSnapshot<>` in many places.

Reply:
Good call -- we use `IOptionsMonitor` elsewhere in the OpenAPI package too so consistency here makes sense.